src/actor/calc.rs

```diff
@@ -0,0 +1,301 @@
+pub mod lookup;
+use super::stat::{SpecialStat, Stat, Stats};
+use crate::sim::SimState;
+use math::round::floor;
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum AttackType {
+    PHYSICAL,
+    MAGIC,
+}
+
+impl Default for AttackType {
+    fn default() -> Self {
+        AttackType::PHYSICAL
+    }
+}
+
+// https://www.akhmorning.com/allagan-studies/how-to-be-a-math-wizard/shadowbringers/damage-and-healing/#direct-damage-d
+pub fn direct_damage(
+    sim: &SimState,
+    potency: i64,
+    job: lookup::Job,
+    stats: &Stats,
+    attack_type: AttackType,
+    multipliers: Vec<f64>,
+) -> i64 {
+    let fatk = attack_power(job, stats.get(Stat::AttackPower));

```

Comment:
The variable names here are particularly terse, and hard to follow as someone who's not terribly familiar with FFXIV.

Reply:
They're intended to be as close to the resource documentation as possible (as seen in the inline comments containing the copied formulas) to make it easy to verify that the translation of the formulas are correct.

`fatk` = `f(ATK)`  
`fdet` = `f(DET)`

Otherwise, I agree completely... fatk is completely meaningless outside of that context.

Reply:
Makes sense; I've seen the same thing in scientific programming.